libpimeval/src/pimParamsDram.h

```diff
@@ -24,6 +24,10 @@ class pimParamsDram
   float getNsAAP() const { return m_tCK * (m_tRAS + m_tRP); }
   double getTypicalRankBW() const { return m_typicalRankBW; }
 
+  double getPjRowRead() const { return m_VDD * (m_IDD0 * (m_tRAS + m_tRP) - (m_IDD3N * m_tRAS + m_IDD2N * m_tRP)); } // Energy for 1 Activate command (and the correspound precharge command) in one subarray of one bank of one chip
+  double getPjLogic() const { return 0.007 * m_tCK * m_tCCD_S ; } // 0.007 mW is the total power per BSLU, 0.007 * m_tCK * m_tCCD_S is the energy of one BSLU during one logic operation in pJ.
```

Comment:
Energy is power multiply by time. Why do we have two time variables here?

Reply:
The 0.007mW is from synthesis result without scaling. We need to check if it is accurate enough

Reply:
I copied that from line 23 of this file; in getNsTCCD(), there are also two time variables multiplied together, I suppose getNsTCCD() should be the timing of one logic micro-ops?

Reply:
> Energy is power multiply by time. Why do we have two time variables here?

@deyuan could you please elaborate what do you mean by this comment? We need to multiply power by time to get the energy.

Reply:
> > Energy is power multiply by time. Why do we have two time variables here?
> 
> @deyuan could you please elaborate what do you mean by this comment? We need to multiply power by time to get the energy.

m_tCK is time unit, m_tCCD_S is also time unit. What do we multiple power by both of them?

Reply:
@deyuan So in the config file, only the m_tCK is the time unit in ns, the other parameters like m_tCCD_S are the number of cycles, they don't have any unit.

Reply:
> @deyuan So in the config file, only the m_tCK is the time unit in ns, the other parameters like m_tCCD_S are the number of cycles, they don't have any unit.

You are right. Sorry I forgot this detail. Thanks for double checking!